tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalCovectorAndMagnitude.cpp

```diff
@@ -0,0 +1,306 @@
+        unnormalized_normal_covectors,
+        db::get<domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                            Frame::Inertial>>(
+            *box));
+
+    if constexpr (UseFlatSpace) {
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          get(magnitude(
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorAndMagnitude<Dim>>(*box)
+                       .at(direction)))));
+    } else {
+      const auto& normal_covector =
+          get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+              *get<evolution::dg::Tags::InternalFace::
+                       NormalCovectorAndMagnitude<Dim>>(*box)
+                   .at(direction));
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          sqrt(get(
+              dot_product(normal_covector, normal_covector,
+                          get<InverseSpatialMetric<Dim>>(fields_on_face)))));
+    }
+
+    for (const double magnitude_of_normal :
+         get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+             *get<evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<
+                  Dim>>(*box)
+                  .at(direction)))) {
+      CHECK(magnitude_of_normal > 0.0);
+    }
+    for (size_t i = 0; i < Dim; ++i) {
+      const auto& normal_covector_and_magnitude =
+          *get<evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<
+              Dim>>(*box)
+               .at(direction);
+      CHECK_ITERABLE_APPROX(
+          unnormalized_normal_covectors.at(direction).get(i),
+          DataVector{
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  normal_covector_and_magnitude)
+                  .get(i) *
+              get(get<evolution::dg::Tags::InternalFace::MagnitudeOfNormal>(
+                  normal_covector_and_magnitude))});
+    }
+  }
+}
+
+template <size_t Dim>
+void test_flat_space(const bool use_moving_mesh) {
+  INFO("Testing flat space");
+  CAPTURE(Dim);
+  CAPTURE(use_moving_mesh);
+  const size_t number_of_grid_points_per_dim = 3;
+  auto box = create_box<Dim>(number_of_grid_points_per_dim, use_moving_mesh);
+  check_normal_covector_quantities<true, Dim>(make_not_null(&box));
+
+  if (use_moving_mesh) {
+    // Mutate the x component of the unnormalized normal vector to simulate
+    // moving mesh
+    db::mutate<ScaleZeroIndex>(
+        make_not_null(&box),
+        [](const gsl::not_null<double*> scale_zero_index) noexcept {
+          *scale_zero_index = 2.3;
+        });
+
+    check_normal_covector_quantities<true, Dim>(make_not_null(&box));
+  }
+}
+
+template <size_t Dim>
+void test_curved_space(const bool use_moving_mesh) {
```

Comment:
This is the same function as the flat space case except for one template parameter being true/false and one word in a print statement. Instead of duplicating 20 lines of code, I would suggest adding a `bool IsFlatSpace` template parameter to a generic test helper.

(I predict you'll say "but the extra template parameter makes it harder to read", to which I counter "yes, but so is reading two 20-line functions several times to see if they're different, only to discover they're the same and end up confused why they weren't a single function")

Reply:
Actually, my reaction is "What was I thinking writing two functions??!!!" :P So the box for me it is!

![image](https://user-images.githubusercontent.com/13531030/100944324-e292e480-34b3-11eb-91ac-5a819f4574dd.png)